packit/patches.py

```diff
@@ -479,6 +480,7 @@ def create_patches(
         """
         files_to_ignore = files_to_ignore or []
         contained = self.are_child_commits_contained(git_ref)
+        current_branch = self.lp.ref
```

Comment:
I think this could be moved in the if-block that follows.

And maybe renamed to `initial_branch`, especially b/c reading `checkout_ref(current_branch)` at the end is pretty confusing.

An even better solution would be to switch back to the initial branch in `linearize_history()`. So that the branch-change is not disconnected and spread in two methods. Put in some other words: in the current solution it's not obvious why the initial branch needs to be saved, as the branch change is hidden in `linearize_history()`.

For the above to work `linearize_history()` could return the branch name it used to do it's thing, and `get_commits_since_ref` and `run_git_format_patch` could use that as a reference.

Reply:
good points, will do, thanks